### PR TITLE
Custom frontmatter preservation, plus series subfolder recursion bug fix

### PR DESCRIPTION
**Pull request by escidmore**
- Frontmatter preservation
  - Added a setting "Preserve custom frontmatter" (default on), saved and migrated (settings version bumped).
  - updateNote now keeps existing YAML frontmatter intact when rewriting the body.
  - Managed vs. custom keys: managed keys are refreshed; stale managed keys are dropped; custom keys are kept when preservation is on and removed when off.
  - Ordering preserved: frontmatter is rebuilt following the original key order; custom keys stay where the user put them; new managed keys append in a deterministic order.
  - Handles spaced/numeric custom keys (e.g., "2025 TBR") without moving or removing them.
- File lookups / sync correctness
  - findNoteByHardcoverId now searches the target folder recursively, so grouped/subfolder notes are found and updated instead of recreated.
- Tests
  - Added coverage for custom frontmatter preservation (including spaced keys), order preservation, stale managed key removal, and nested-folder note lookup.

### CODE REVIEW

File: src/services/NoteService.ts [hunk 1]
```diff
@@ -450,6 +459,92 @@ export class NoteService {
...
+	private extractFrontmatter(content: string): {
+		frontmatterText: string;
+		bodyText: string;
+	} {
+		const match = content.match(/^---\n[\s\S]*?\n---\n/);
+		if (match) {
+			const frontmatterText = match[0];
+			const bodyText = content.slice(frontmatterText.length);
+			return { frontmatterText, bodyText };
+		}
```
Comment (aliceinwaterdeep) [inline src/services/NoteService.ts]: I previously received feedback from the Obsidian team about not using regex for frontmatter. However, I believe your approach here should be fine since you're not parsing YAML properties. Just adding a note for the future in case it gets challenged by reviewers.

**Review by aliceinwaterdeep — COMMENTED** [review]
Thank you so much for this PR! This is excellent work, I really appreciate how thoughtful and thorough you've been (and the tests!).

I've been testing it today and everything works great! I'm happy to merge it, but I found one edge case that needs addressing: renamed managed keys create duplicates.

For example:
- User has `authors: ["Elena Ferrante"]`
- User renames the property to `bookAuthors` in settings
- the old property gets preserved as custom
- Next sync: Note has both `authors` and `bookAuthors` with the same data

Without custom property preservation, this wasn't an issue as the old property would just get wiped.

The only fix I have in mind is using value comparison: when a key is no longer managed, we check if its value matches any new managed values. If so, it's likely a rename and shouldn't be preserved. It's not great but I can't think of another way around it. 
The only extremely edge case is if a true custom property holds the same value as a "previously managed" property but I think the likelihood of that happening is very low.
I'd love to know your thoughts on this!

**Comment by escidmore** [comment]
Sorry for a bit of a delay in responding; I got caught up in some other projects. For the renamed property thing I can think of three different possible solutions, each with pros and cons. I'm happy to try to implement any of them you might prefer.

---
**1. Value-Based Duplicate Detection (your suggestion)**
Pro:
- No extra state tracking or storage is required
- It works retroactively for existing users
- It's easy

Con:
- The edge case you mentioned where it might remove something that was actually custom 
---
**2. Track Historical Property Names**
Store a persistent list of all property names ever managed by the plugin. When checking for preservation, exclude anything in this historical set.

Pro:
- Deterministic
- Won't remove true custom properties unless there's an even edgier edge case where the custom property happens to match a previously used property name. Although that might actually be more likely, now that I think about it.

Con:
- Requires new persistent storage that grows forever (but probably tiny, unless someone is truly, incredibly indecisive)
- It won't help existing users who already renamed properties
---
**3. Explicit Rename Migration**
When a property name changes in settings, trigger a migration that updates all existing notes to use the new name before the next sync (probably by prompting the user to do so with a button press, optional)

Pro:
- Fixes the data at the source

Con:
- Most complicated to implement (but I offered to do it)
- Need to handle partial migrations if interrupted
---

We could also (for any of these) implement an optional changelog for syncs, which would at least let someone see what was changed, and even have a "rollback" option to undo the things in that changelog in the case where it clobbered some data. Also something I'm willing to implement, if you want.

**Comment by aliceinwaterdeep** [comment]
Please no need to apologize, I appreciate you taking the time to think about solutions! They all have their merits and I agree that option 3 would be the "proper" way to handle it. However, the complexity is probably not worth it right now and would give us even more cases to handle (e.g. what if a migration fails midway?). 
I'd save option 3 for when/if users are encountering issues that require a migration, or new features are added that could benefit from such a system. For now, I'd say option 1 works fine. Let me know if you want to work on it, otherwise I'm happy to take it myself if you're busy!

**Comment by aliceinwaterdeep** [comment]
I'm merging this now and will implement the fixes we discussed in a follow-up PR to keep things moving. Thanks again, really appreciate your contribution!
